common/djangoapps/enrollment/tests/test_api.py

```diff
@@ -230,3 +230,72 @@ def test_caching(self):
         # The data matches
         self.assertEqual(len(details['course_modes']), 3)
         self.assertEqual(details, cached_details)
+
+    def test_update_enrollment_expired_mode_with_error(self):
+        """ Verify that if verified mode is expired then enrollment cannot be updated if
+        include_expired flag is false."""
+
+        # Add fake course enrollment information to the fake data API
+        fake_data_api.add_course(self.COURSE_ID, course_modes=['honor', 'verified', 'audit'])
+        # Enroll in the course and verify the URL we get sent to
+        result = api.add_enrollment(self.USERNAME, self.COURSE_ID, mode='audit')
+        get_result = api.get_enrollment(self.USERNAME, self.COURSE_ID)
+        self.assertEquals(result, get_result)
+
+        result = api.update_enrollment(self.USERNAME, self.COURSE_ID, mode='honor')
+        self.assertEquals('honor', result['mode'])
+
+        # set the verified mode as expired
+        fake_data_api.set_expired_mode()
+        # On updating enrollment mode to verified it should the raise the error.
+        with self.assertRaises(CourseModeNotFoundError):
+            api.update_enrollment(self.USERNAME, self.COURSE_ID, mode='verified', include_expired=False)
+
+    def test_update_enrollment_expired_mode(self):
+        """ Verify that if verified mode is expired then enrollment can be updated if
+        include_expired flag is true."""
+
+        # Add fake course enrollment information to the fake data API
+        fake_data_api.add_course(self.COURSE_ID, course_modes=['honor', 'verified', 'audit'])
+        # Enroll in the course and verify the URL we get sent to
+        result = api.add_enrollment(self.USERNAME, self.COURSE_ID, mode='audit')
+        get_result = api.get_enrollment(self.USERNAME, self.COURSE_ID)
+        self.assertEquals(result, get_result)
+
+        result = api.update_enrollment(self.USERNAME, self.COURSE_ID, mode='honor')
+        self.assertEquals('honor', result['mode'])
+
+        # set the verified mode as expired
+        fake_data_api.set_expired_mode()
```

Comment:
Don't copy and paste the first section of this test method; factor it out and share it with `test_update_enrollment_expired_mode_with_error` above.